### PR TITLE
Moved mlflow/shap.py to shap/__init__.py for consistency. Fixes #9743

### LINKED ISSUE #9743: Move mlflow/shap.py to mlflow/shap/__init__.py
Related to #9731 

The motivation is to keep each flavor module in its own directory for consistency. For now we just move it to `__init__.py` for simplicity, but in the future we will think about a better refactoring.

Please check #9692 for a similar PR.

Code pointer: [link](https://github.com/mlflow/mlflow/blob/master/mlflow/shap.py)

If you are using VScode, the file change should be automatically tracked. If the moving is not tracked correctly, you can use `git mv` to do the renaming.

### PR DESCRIPTION
**Pull request by issamarabi**
### Related Issues/PRs

Resolve #9743

### What changes are proposed in this pull request?

Moved the SHAP integration code from `mlflow/shap.py` to a directory-based structure at `mlflow/shap/__init__.py` to maintain consistency with other modules in MLflow. This change aligns with the project's direction to modularize flavor integrations.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by chenmoneygithub** [comment]
@issamarabi thanks for the PR! It's my bad, but could you move it to `mlflow/shap/__init__.py` instead? thanks!

File: shap/__init__.py
Comment (harupy) [inline shap/__init__.py:1]: `mlflow/shap/__init__.py`?

**Comment by issamarabi** [comment]
@chenmoneygithub I've made the requested changes and moved the contents to mlflow/shap/__init__.py. Please review again. Thanks!

**Review by harupy — APPROVED** [review]
LGTM！

**Pull request merged**